crates/ruma-events/src/room/server_acl.rs

```diff
@@ -45,14 +46,40 @@ impl RoomServerAclEventContent {
     pub fn new(allow_ip_literals: bool, allow: Vec<String>, deny: Vec<String>) -> Self {
         Self { allow_ip_literals, allow, deny }
     }
+
+    /// Returns true if and only if the server is allowed by the ACL rules.
+    pub fn is_allowed(&self, server_name: &ServerName) -> bool {
+        if !self.allow_ip_literals && server_name.is_ip_literal() {
+            return false;
+        }
+
+        for deny in &self.deny {
+            if let Ok(d) = glob::Pattern::new(deny) {
+                if d.matches(server_name.as_str()) {
+                    return false;
+                }
+            }
+        }
+
+        for allow in &self.allow {
+            if let Ok(a) = glob::Pattern::new(allow) {
+                if a.matches(server_name.as_str()) {
+                    return true;
+                }
+            }
+        }
+
+        false
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use serde_json::{from_value as from_json_value, json};
 
-    use super::RoomServerAclEventContent;
+    use super::*;
     use crate::StateEvent;
+    use ruma_identifiers::server_name;
```

Comment:
This should be in the external crate imports group above, next to the `serde_json` import.